### ISSUE TITLE
Next.js example implement preview mode

### ISSUE THREAD

**Issue by fmontes** [dotCMS/core#21807]
**Is your feature request related to a problem? Please describe.**
Tipically users that create content want to see how that content looks like in their site before pressing "publish" button.

Out of the box Next.js will always fetch the dotCMS live data to build the pages but some cases (preview) we need to fetch "working data"

**Describe the solution you'd like**
Next.js have a solution for this and is "preview mode", all the information [here](https://nextjs.org/docs/advanced-features/preview-mode) 

In short, when preview mode is `true` we need to build the page using the `working` version of the contentlet in dotCMS.

**Additional context**
We need to find a way to provide the best content editor experience as well as the best developer experience in this approach.
